hbase/src/test/java/org/apache/metamodel/hbase/InsertRowTest.java

```diff
@@ -0,0 +1,390 @@
+     * NOTE: This exception is already prevented when using the {@link HBaseRowInsertionBuilder}
+     * @throws IOException 
+     */
+    @Test
+    public void testNotSettingEnoughValues() throws IOException {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+            final LinkedHashMap<HBaseColumn, Object> row = createRow(existingTable, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR, false);
+            final List<HBaseColumn> columns = getHBaseColumnsFromRow(row);
+            final Collection<Object> values = getToLittleValues(row);
+            final HBaseClient hBaseClient = ((HBaseDataContext) getUpdateCallback().getDataContext()).getHBaseClient();
+            hBaseClient.insertRow(TABLE_NAME, columns.toArray(new HBaseColumn[columns.size()]), values.toArray(
+                    new Object[values.size()]), 0); // TODO: find the ID-column
+            fail("Should get an exception when insering directly into the HBaseClient without having enough values.");
+        } catch (IllegalArgumentException e) {
+            assertEquals("The amount of columns don't match the amount of values", e.getMessage());
+        }
+    }
+
+    /**
+     * Goodflow. Using an existing table and columns, should work
+     */
+    @Test
+    public void testInsertIntoWithoutExecute() {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+            final LinkedHashMap<HBaseColumn, Object> row = createRow(existingTable, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR, false);
+            final List<HBaseColumn> columns = getHBaseColumnsFromRow(row);
+            getUpdateCallback().insertInto(existingTable, columns);
+        } catch (Exception e) {
+            fail("No exception should be thrown, when inserting into an existing table.");
+        }
+    }
+
+    /**
+     * Goodflow, creating a row with qualifiers null should work.
+     */
+    @Test
+    public void testQaulifierNull() {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+            final LinkedHashMap<HBaseColumn, Object> row = createRow(existingTable, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR, true);
+            final List<HBaseColumn> columns = getHBaseColumnsFromRow(row);
+
+            checkRows(false, true);
+            final HBaseRowInsertionBuilder rowInsertionBuilder = getUpdateCallback().insertInto(existingTable, columns);
+            setValuesInInsertionBuilder(row, rowInsertionBuilder);
+            rowInsertionBuilder.execute();
+            checkRows(true, true);
+        } catch (Exception e) {
```

Comment:
Same like above, don't catch the exception but have the method throw `IOException` instead.